### PR TITLE
add delete button without refresh page it's coming later

### PR DESCRIPTION
**Pull request by FreytagDom**

### CODE REVIEW

File: components/SavedData.js [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+    const response = await fetch('/api/savedInsulinData', {
+      method: 'DELETE',
+      headers: {
+        'content-type': 'application/json',
+      },
+
+      body: JSON.stringify(data),
+    });
+
+    const json = response.json();
```
Comment (NikRoe) [inline components/SavedData.js:11]: do you need this at a later point? For the moment, it is not being used

File: pages/api/savedInsulinData/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    case 'GET':
+      const savedInsulinData = await getAllCategories();
+      return response.status(200).json({ savedInsulinData });
+
+    case 'DELETE':
+      await dbConnect();
+
+      const data = request.body;
+
+      await SavedInsulinData.findOneAndDelete();
```
Comment (NikRoe) [inline pages/api/savedInsulinData/index.js]: you declare the variable `data` but don't use it anywhere, why?

how does your backend know which item to delete? 

you have some better logic and another delete handler in your dynamic api route `[id].js`. You can delete the whole DELETE case.

File: components/SavedData.js [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
 
+  async function removeCard() {
+    const response = await fetch('/api/savedInsulinData', {
+      method: 'DELETE',
+      headers: {
+        'content-type': 'application/json',
+      },
+
+      body: JSON.stringify(data),
+    });
```
Comment (NikRoe) [inline components/SavedData.js]:
```suggestion
  async function removeCard(id) {
    const response = await fetch('/api/savedInsulinData/' + id, {
      method: 'DELETE',
    });
```

File: components/SavedData.js [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
+  }
   return (
     <WrapperSaved>
       <CardGrid>
         {data.map((item) => {
           return (
-            <Saved key={item.id}>
+            <Saved key={item.id} removeCard={removeCard}>
+              <IconWrapper>
+                <RiDeleteBinLine onClick={() => removeCard(item.id)} />
```
Comment (NikRoe) [inline components/SavedData.js]: your `Saved` component does not need the prop `removeCard`.
```suggestion
            <Saved key={item.id}>
              <IconWrapper>
                <RiDeleteBinLine onClick={() => removeCard(item.id)} />
```

File: pages/api/savedInsulinData/[id].js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { getCategoryById } from '../../../services/insulinAppDataService';
+import SavedInsulinData from '../../../models/InsulinApp';
+
+export default async function handler(request, response) {
+  const { id } = request.query;
+  switch (request.method) {
+    case 'GET':
+      const cardData = getCategoryById(id);
```
Comment (NikRoe) [inline pages/api/savedInsulinData/[id].js:8]: the naming seems a bit off. You store the results of a function called `getCategoryById()` in a variable called `cardData`. This is confusing.

File: pages/api/savedInsulinData/index.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+      return response
+        .status(201)
+        .json({ message: 'SavedInsulinData was created' });
+  }
```
Comment (NikRoe) [inline pages/api/savedInsulinData/index.js]: You don't need the DELETE case here and could go with a simple `if...else`:

```suggestion
if(request.method === "GET"){
      const savedInsulinData = await getAllCategories();
      response.status(200).json({ savedInsulinData });
      } else {
      response.status(405).json({message: "Method not allowed"})
      };
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hey Dominic,

looks great overall, however, there are some changes to your API route you should consider implementing.

Keep it up! 🙌

File: pages/api/savedInsulinData/index.js (hunk 6 above)
Comment (FreytagDom) [reply pages/api/savedInsulinData/index.js]: Thanks for the good feedback, I have changed a few things.

File: components/SavedData.js [hunk 7]
```diff
@@ -1,14 +1,24 @@
 import styled from 'styled-components';
+import { RiDeleteBinLine } from 'react-icons/ri';
 
 export default function SavedDataInjected({ cardData }) {
   const data = cardData;
-
+  console.log(cardData);
```
Comment (NikRoe) [inline components/SavedData.js]: you can remove this console.log

**Review by NikRoe — APPROVED** [review]
Hi Dominic,

looks good! Only one left over console.log you should remove before going to QA.

**Pull request merged**